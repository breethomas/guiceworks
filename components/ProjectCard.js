// @flow
import React from 'react'
import css, { before, hover, media2, media3, media4 } from '../styles/css'
import Link from './Link'
import type { ProjectProps } from '../types/app.js.flow'

type Props = {
  project: ProjectProps,
}

const style = css(
  {
    display: 'flex',
    flexFlow: 'column wrap',
    alignItems: 'center',
    justifyContent: 'center',
    position: 'relative',
    width: '100%',
    height: '100vw',
    maxHeight: 350,
    textDecoration: 'none',
    backroundRepeat: 'none',
    backgroundSize: 'cover',
  },
  before({
    content: '""',
    position: 'absolute',
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    backgroundColor: 'rgba(0, 0, 0, 0)',
    transition: 'background-color 0.4s',
  }),
  hover(before({
    backgroundColor: 'rgba(0, 0, 0, 0.7)',
  })),
  media2({ width: '50%' }),
  media3({ width: '33.33333%' }),
  media4({ width: '25%' }),
)

const textStyle = css({
  marginTop: 40,
  opacity: 0,
  fontSize: 14,
  textTransform: 'uppercase',
})

export default ({ project, ...props }: Props) => (
  <Link
    className={style}
    as={`/projects/${project.id}`}
    href={`/projects?id=${project.id}`}
    key={project.title}
    style={{ backgroundImage: `url(${project.image})` }}
    {...props}
  >
    <span className={textStyle}>{project.title}</span>
  </Link>
)

