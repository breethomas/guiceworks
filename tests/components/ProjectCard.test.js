import React from 'react'
import renderer from 'react-test-renderer'
import ProjectCard from '../../components/ProjectCard'

describe('ProjectCard', () => {
  let component

  beforeEach(() => {
    const project = {
      body: 'Project 1 _body_',
      icon: 'icon.svg',
      id: 'project-1',
      image: '/static/images/image_home.jpg',
      greybox: '/static/images/image_grey-box.jpg',
      next: 'next-project',
      prev: 'prev-project',
      title: 'Project 1 title',
    }

    component = (
      <ProjectCard
        project={project}
      />
    )
  })

  it('shows the correct snapshot for the ProjectCard component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
