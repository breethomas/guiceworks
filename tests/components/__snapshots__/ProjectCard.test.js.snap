// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard shows the correct snapshot for the ProjectCard component tree 1`] = `
<a
  className={
    Object {
      "data-css-19armf": "",
    }
  }
  href="/projects/project-1"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(/static/images/image_home.jpg)",
    }
  }
>
  <span
    className={
      Object {
        "data-css-fv2zf4": "",
      }
    }
  >
    Project 1 title
  </span>
</a>
`;
